### DISCUSSION TITLE
How to get raw request path in middleware?

### DISCUSSION THREAD

**Discussion by CaptainN** [vercel/next.js#37754] [Help]
Is there a way to get the raw request pathname in middleware? I specifically want the full pathname with locale - for example, `/en/about-us`. By default, the next API only grants access to the transformed pathname - that is, it'll say `/about-us` for a locale prefixed route.

This is all I need to solve the inconsistent prefix problem for the defaultLocale. Without that though, I can't detect if I'm on a prefixed route or a prefixless route.

**Comment by vikingair** [comment]
How about this?

```ts
const fullPath = `/${request.nextUrl.locale}${request.nextUrl.pathname}`
```

Or to be more precise:

```ts
const localePrefix = request.nextUrl.locale === defaultLocale ? '' : '/' + request.nextUrl.locale;
const fullPath = localePrefix + request.nextUrl.pathname;
```

**Reply by icyJoseph** [reply]
> `defaultLocale` comes from `request.nextURL.defaultLocale`

I think this ought to work!

**Reply by wereHamster** [reply]
@fdc-viktor-luft this is not correct if the request is for `/{defaultLocale}/{...rest}`, eg. I have default locale `de` and a request to `/de/foo` would transform to `/foo` fullPath in your code.

**Reply by vikingair** [reply]
I've experienced that NextJS internally redirects any requests to `/de/foo` into `/foo` if `de` is the configured default locale. I assume that this happens before the middleware code gets invoked.

**Comment by wereHamster** [comment]
I opened a PR, hopefully we can get this resolved before the middleware API is set in stone… #37885
